sti-main/src/main/java/uk/ac/shef/dcs/sti/core/algorithm/tmp/TMPOdalicInterpreter.java

```diff
@@ -71,6 +73,26 @@ public TAnnotation start(Table table, Constraints constraints) throws STIExcepti
           .compute(table, constraints.getSubjectColumnPosition(), ignoreColumnsArray);
       tableAnnotations.setSubjectColumn(subjectColumnScores.get(0).getKey());
       
+      // when the column does not contain Named entity as the most frequent data type,
+      // we will not disambiguate (and so classify) them, except for
+      // the user defined constraints
+      Set<Ambiguity> newAmbiguities = new HashSet<>(constraints.getAmbiguities());
```

Comment:
Hi,
this is solution to a bug #308 , which we discussed on Skype. It may be a bit dirty, so I am open to any suggestions, it might be best to discuss it at Skype at least, with @josef-janousek . Here is the conversation that we had with @josef-janousek about it (ignore the part about relations):

> [11.1.2017 11:18:53] Václav Brodec: vše daultní, zkoušel jsem třídy v hlavičce
[11.1.2017 11:19:06] Václav Brodec: tedy kódování iso-8859-1
[11.1.2017 11:19:11] Václav Brodec: mimo to
[11.1.2017 11:19:35] Josef Janoušek: a počet řádek?
[11.1.2017 11:19:46] Václav Brodec: výchozí
[11.1.2017 11:19:55] Josef Janoušek: a jak dlouho to trvalo?
[11.1.2017 11:20:15] Václav Brodec: měl jsem nacachováno, byla to ta verze souboru s deseti řádky
[11.1.2017 11:20:19] Václav Brodec: takže chvíli
[11.1.2017 11:20:58] Josef Janoušek: dobře, zkusím
[11.1.2017 11:49:17] Josef Janoušek: tak možná už jsem na něco přišel, ještě prosímtě na jakých třídách konkrétně jsi to zkoušel?
[11.1.2017 11:50:12] Václav Brodec: zkusil jsem si třeba navrhnout vlastní  třídu pro district
[11.1.2017 11:50:50] Josef Janoušek: district_name nebo district_code ?
[11.1.2017 11:56:09] Václav Brodec: code
[11.1.2017 11:59:10] Josef Janoušek: ano, tak to odpovídá, zjistil jsem totiž, že např. na district_name mi to funguje v pořádku, a na district_code (nebo sloupec livebirths) to mizí, tedy mizí to ve sloupcích, které nemají žádný výsledek disambiguace a možná to souvisí i s tím, že to jsou sloupce pouze s číselnými hodnotami
[11.1.2017 12:05:21] Václav Brodec: ok, díky
zkusíš s tím tedy něco udělat?
[11.1.2017 12:06:37] Josef Janoušek: zkusím, no, ale nevím, jak to půjde
[11.1.2017 12:21:35] Josef Janoušek: a k tomu zadávání relací: manuální přidávání relací mi funguje všude v případě, že jsou přidávány z vyhledávání; 
propose nových relací funguje dobře v případě, že jsou dané třídy klasifikovány;
ale v případě, kdy dané třídy klasifikovány nejsou, tak nejsou v propose dialogu vyplněny hodnoty domain a range a tím pádem nelze novou relaci proposovat (což je ale asi celkem logické)
[11.1.2017 12:22:31] Václav Brodec: ano, toho jsem si všiml, a zadal jsem pro to issue, neboť jsme se domlouvali, že domain a range by měly být nepovinné
[11.1.2017 12:25:35] Josef Janoušek: tak já nevím přesně, jak to tam funguje, ale vzhledem k tomu, že relace je, jak jsem pochopil, vztah mezi třídami, tak je otázka, co by takový vztah mezi neklasifikovanými sloupci potom znamenal (do RDF exportu by se pravděpodobně stejně nic nepropsalo)
[11.1.2017 12:26:37] Václav Brodec: no a není to tak, že se ta relace užije i pro jednotlivé buňky?
[11.1.2017 12:28:00] Josef Janoušek: aha, ano, ve výsledku by se měla užít pro propojení jednotlivých buněk, tak to jo
[11.1.2017 17:03:16] Josef Janoušek: takže, už jsem na to přišel: je to způsobeno těmito řádky: https://github.com/odalic/sti/blob/develop/sti-main/src/main/java/uk/ac/shef/dcs/sti/core/algorithm/tmp/TMPOdalicInterpreter.java#L83-L84 , které způsobí, že pokud algoritmus nějakým způsobem zjistí, že daný sloupec neobsahuje Named Entity, tak se tento sloupec už vůbec neklasifikuje ani nedisambiguuje (stejně jako by byl ignorován); takže pokud tyto řádky zakomentuješ, můžeš si zkusit, jaké by to bylo bez toho - a v tom případě u
[11.1.2017 17:03:32] Josef Janoušek: v tom případě již bude i feedback fungovat
[11.1.2017 17:04:43] Josef Janoušek: ovšem na druhou stranu asi je k tomu nějaký dobrý důvod, že se tyto sloupce, které neobsahují named entity už neposílají ke zpracování, stejně by se totiž asi nic moc nenašlo... když jsem to zkoušel, nelezené výsledky nebyly příliš správné
[11.1.2017 17:11:27] Josef Janoušek: ještě konkrétně u toho souboru Narození, když jsem dané řádky zkusil zakomentovat, aby se tedy zpracovaly i sloupce co neobsahují named entity, tak došlo k nějaké chybě ve SPARQLProxy; takže ještě je druhá možnost, jak toto zkusit, a to, že se tyto sloupce nastaví jako povinné, to můžeš udělat na řádku https://github.com/odalic/sti/blob/develop/sti-main/src/main/java/uk/ac/shef/dcs/sti/core/algorithm/tmp/TMPOdalicInterpreter.java#L35 zápisem super(new int[0], new int[] {2, 3, 5, 7, 8}); přičemž ale ta chyba nastává ve sloupci č. 7, takže pokud ten vynecháš, čili super(new int[0], new int[] {2, 3, 5, 8}); tak budou ty sloupce 2,3,5,8 povinné a i feedback na nich bude fungovat
[11.1.2017 17:14:05] Václav Brodec: no a nešlo by takový případ převést na to, že sice sloupec budeš klasifikovat, ale zakážeš disambiguace v těch buňkách? tedy hint pro klasfiikaci se užije?
[11.1.2017 17:16:19] Josef Janoušek: možná by to šlo, zkusím; 
feedback pro disambiguace tedy rovněž fungovat v těchto případech nebude
[11.1.2017 17:45:10] Václav Brodec: jasně, ale tak opět by tam šlo zakázat tu disambiguaci v těch buňkách ,kde jsem neposkytl feedback
>
>i pokud to takhle půjde, tak tenhle postup zkus popsat v tom kódu nějak transparentně, aby to nebyl nějaký zasunutý hack
[11.1.2017 17:48:53] Josef Janoušek: takže chování by bylo takové, že pokud jsem poskytl feedback, tak se použije, a jinak se nebude nic vyhledávat v KB (a to by platilo jak pro klasif., tak pro disamb.) ?
[11.1.2017 17:49:33] Václav Brodec: no jenom v těch případech, kdy by došlo k tomu ignorování jak jsi říkal
[11.1.2017 17:50:55] Josef Janoušek: ano, toto chování by platilo pouze pro sloupce, kde algoritmus rozpoznal, že neobsahují named entity
[11.1.2017 17:55:23] Václav Brodec: ok, zkus
>
>asi to je jasně, ale pro jistotu: tyhle machinace by se neměly nijak projevit navenek, tedy to, že jsme ty disambiugace zakázali nekde, kde by byla zakázána automaticky, tak pak nemodifikovat ten uživ. feedback
[11.1.2017 18:00:55] Josef Janoušek: asi bych si za tím účelem zmodifikoval pouze Constraints ve sti-main (které jsou nyní také immutable, takže je asi budu potřebovat nemít immutable, nebo mám zkonstruovat novou instanci?), takže uživatelský feedback by to ovlivnit nemělo
[11.1.2017 18:04:02] Václav Brodec: ok, no pokud to zvládneš tak, že vytvoříš na základě té verze odvozené z Feedbacku novou, která bude mít tyhle změny (a nebo budeš tuhle úpravu řešit třeba až na úrovni toho algoritmu - v okamžiku, kdyb udeš vědět, co bude přesně ignorováno), tak zkus ty Constraints ponechat immutable
[11.1.2017 18:14:35] Josef Janoušek: taky si tedy říkám, že je to takové poněkud neprůhledné, tak jestli to ještě nenechat na další iteraci a tam to vyřešit nějak komplexněji (např. i s informováním uživatele)...
[11.1.2017 18:15:20] Václav Brodec: no já osobně si myslím, že by bylo lepší to vyřešit úpravou toho algoritmu, byť třeba podle toho nápadu s tím zakázáním
[11.1.2017 18:16:08 | Edited 18:16:30] Václav Brodec: pokud máš pocit, že to nejseš schopen čistě napsat během dneška, tak to samozřejmě posuneme
>
> nemá smysl tam něco nabouchat, abys to pozítří začal přepisovat
[11.1.2017 18:24:29] Josef Janoušek: tak dneska to ještě zkusím, spíše mám obavu, aby se tím, jak budu upravovat ty constraints, případně zase něco jiného nerozbilo
>
> obecnější otázka potom je, jestli tyto "non-named-entity" tedy skutečně vůbec nezpracovávat, nebo zpracovávat nebo řešit ještě nějak jinak
[11.1.2017 21:31:30] Josef Janoušek: Ahoj, tak už jsem to dokončil, je to v nové větvi, takže můžeš prohlédnout a otestovat